src/components/SingleEvent/Admin/PartyForm.js

```diff
@@ -485,35 +488,87 @@ class PartyForm extends Component {
                   }
+                  return (
+                    <>
+                      <InputWrapper>
+                        <Label>Currency</Label>
+                        <CurrencyPicker
+                          currencyType={this.state.currencyType}
+                          onChange={currencyType => {
+                            let tokenAddress
+                            if (currencyType === 'ETH') {
+                              tokenAddress = EMPTY_ADDRESS
+                            } else if (currencyType === 'DAI') {
+                              tokenAddress = this.state.daiAddress
+                            } else {
+                              tokenAddress = ''
+                            }
+                            console.log(currencyType)
+                            this.setState({
+                              currencyType,
+                              tokenAddress,
+                              deposit: 0
+                            })
+                          }}
+                        />
+                      </InputWrapper>
+                      {this.state.currencyType === 'TOKEN' && (
+                        <InputWrapper>
+                          <Label>Token Address</Label>
+                          <TextInput
+                            value={this.state.tokenAddress}
+                            onChangeText={val => {
+                              this.setState({ tokenAddress: val })
+                            }}
+                            type="text"
+                            placeholder="0x..."
+                          />
+                        </InputWrapper>
+                      )}
+                    </>
+                  )
+                }}
+              </SafeQuery>
+              <SafeQuery query={TOKEN_QUERY} variables={{ tokenAddress }}>
+                {({
+                  data: {
+                    token: { name, symbol, decimals }
+                  },
+                  loading
+                }) => {
                   return (
                     <InputWrapper>
-                      <Label>Currency</Label>
-                      <CurrencyPicker
-                        currencyType={this.state.currencyType}
-                        daiAddress={this.state.daiAddress}
-                        onChange={({ currencyType, tokenAddress }) => {
-                          this.setState({ currencyType, tokenAddress })
+                      <Label>Commitment</Label>
+                      <CommitmentInput
+                        value={deposit}
+                        onChangeText={val => {
+                          let regex = /.*/
+                          if (decimals === 0 || decimals === '0') {
+                            regex = new RegExp(`^\\d*$`)
+                          } else if (decimals > 0) {
+                            regex = new RegExp(`^\\d*(\\.\\d{0,${decimals}})?$`)
```

Comment:
Please rename variable to something more descriptive. And possibly add a comment if that's not enough to describe what this is doing without having to understand the regex